### PR TITLE
Allow table data to download as Excel

### LINKED ISSUE #3822: [Feature] Allow downloading data from a table as an Excel file
The table widget today can already export data as a CSV file, but importing CSV files into Excel is a painful process. Setting the encoding and CSV formatting settings when importing is a fragile and error-prone process.

So, if the table widget can export an Excel file directly, it'd be a lot more convenient export format to work with.

### PR DESCRIPTION
**Pull request by vicky-primathon**
## Description
Allow table data to download as Excel

Fixes #3822 

## Type of change
- New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: feature/table-excel-export 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 48.94 **(0)** | 28.71 **(0.01)** | 26.78 **(0)** | 49.32 **(0)**
 :green_circle: | app/client/src/components/designSystems/appsmith/TableComponent/TableStyledWrappers.tsx | 35.8 **(0.43)** | 16.95 **(0.14)** | 0 **(0)** | 35.8 **(0.43)**</details>

### CODE REVIEW

**Review by somangshu — CHANGES_REQUESTED** [review]
@vicky-primathon the package you have used `xlxs` is very heavy for the use case we have right now. I suggest we use like [zipcelx](https://github.com/egeriis/zipcelx), which is lightweight and performant.

**Review by somangshu — DISMISSED** [review]
Approval for running test

**Review by somangshu — APPROVED** [review]

**Comment by somangshu** [comment]
Tested on DP

**Comment by marks0351** [comment]
@vicky-primathon the excel version seems to have extra empty columns and rows if some table columns are hidden.
Screenshot 2021-05-06 at 6 51 36 PM

**Comment by marks0351** [comment]
Just a suggestion, I feel like the download options are disabled coz of the color. make sure u guys run it by UX.
Screenshot 2021-05-07 at 5 42 36 PM

**Review by marks0351 — APPROVED** [review]

**Comment by somangshu** [comment]
> Just a suggestion, I feel like the download options are disabled coz of the color. make sure u guys run it by UX.
> Screenshot 2021-05-07 at 5 42 36 PM

@momcilo-appsmith can you help us out with this?

**Review by somangshu — CHANGES_REQUESTED** [review]
@vicky-primathon the developer handoff design is quite different from what you have made as @marks0351 has pointed out. Kindly check the design and specs for more details
Screenshot 2021-05-10 at 2 27 47 PM

File: app/client/src/components/designSystems/appsmith/TableComponent/TableAction.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    background: ${Colors.ATHENS_GRAY};
+  }
+`;
+
+function TableAction(props: TableActionProps) {
+  return (
+    <TableIconWrapper
+      className={props.className}
+      onClick={(e) => {
+        props.selectMenu(!props.selected);
```
Comment (riodeuno) [inline app/client/src/components/designSystems/appsmith/TableComponent/TableAction.tsx:43]: Let's useCallback for this.

**Review by riodeuno — CHANGES_REQUESTED** [review]
Some comments to remove inline arrow function props.

**Review by riodeuno — APPROVED** [review]

**Pull request merged**